test/integration/helpers_test.go

```diff
@@ -183,3 +187,34 @@ func gatewayLinkStatusMatches(t *testing.T, c *gatewayclient.Clientset, verifyLi
 	// the link existence return false
 	return !verifyLinked
 }
+
+// setIngressClassNameWithRetry changes Ingress.Spec.IngressClassName to specified value
+// and retried if update confilict happens.
```

Comment:
```suggestion
// and retries if the update fails.
```

Reply:
updated, this `retry` only retry on update conflicts but not other errors.